omou-frontend/src/reducers/calenderReducer.js

```diff
@@ -0,0 +1,44 @@
+import initialState from './initialState';
+import * as actions from "../actions/actionTypes"
+
+export default function calender(state = initialState.CalenderData, { payload, type }) {
+    let newState = state;
+    let obj = newState.events_in_view;
+    switch (type) {
+        case actions.ADD_EVENT:
+            newState.events_in_view.push(payload);
+            console.log('CalenderData Action', payload, newState);
+            return state;
+
+        case actions.DELETE_EVENT:
+            for (let i = 0; i < newState.events_in_view.length; i++) {
+                let obj = newState.events_in_view[i];
+                if (payload.indexOf(obj.id) !== -1) {
+                    newState.events_in_view.splice(i, 1);
```

Comment:
When you do this, this will always skip over 1 element because the list is getting shorter at the index where you're removing something. I'm also a bit confused - isn't the payload the event object you want to delete?  How is payload an array? make sure to rename this to something more clear. 

let me know if your logic is correct - you're iterating through each event in view. If it is contained in your payload array, then you remove it. I'm assuming your payload is an array of the events you want to delete? I'm not sure why you'd have an array of events you'd want to delete when you call `deleteEvent()` . If you're looking to delete a specific event object from the `events_in_view` array, try [this answer from github](https://gist.github.com/scottopolis/6e35cf0d53bae81e6161662e6374da04).